src/python/bot/fuzzers/libfuzzer.py

```diff
@@ -413,6 +414,201 @@ class UnshareLibFuzzerRunner(new_process.UnshareProcessRunnerMixin,
   """LibFuzzerRunner which unshares."""
 
 
+class FuchsiaUndercoatLibFuzzerRunner(new_process.UnicodeProcessRunner,
+                                      LibFuzzerCommon):
+  """libFuzzer runner (when Fuchsia is the target platform, and undercoat
+  is used)."""
+
+  def __init__(self, executable_path, default_args=None):
+    # We always assume QEMU is running on __init__, since build_manager sets
+    # it up initially.
+    # Note: In this case executable_path is simply `package/fuzzer`
+    super().__init__(executable_path=executable_path, default_args=default_args)
+    self.handle = environment.get_value('FUCHSIA_INSTANCE_HANDLE')
+
+  def _corpus_directories_libfuzzer(self, corpus_directories):
+    """ Returns the corpus directory paths expected by libfuzzer itself. """
+    return [
+        self._target_corpus_path(os.path.basename(corpus_dir))
+        for corpus_dir in corpus_directories
+    ]
+
+  def _new_corpus_dir_host(self, corpus_directories):
+    """ Returns the path of the 'new' corpus directory on the host. """
+    return corpus_directories[0]
+
+  def _new_corpus_dir_target(self, corpus_directories):
+    """ Returns the path of the 'new' corpus directory on the target. """
+    return self._target_corpus_path(
+        os.path.basename(self._new_corpus_dir_host(corpus_directories)))
+
+  def _target_corpus_path(self, corpus_name):
+    """ Returns the path of a given corpus directory on the target. """
+    return 'data/corpus/' + corpus_name
+
+  def _push_corpora_from_host_to_target(self, corpus_directories):
+    # Push corpus directories to the device.
+    self._clear_all_target_corpora()
+    logs.log('Push corpora from host to target.')
+    for corpus_dir in corpus_directories:
+      # TODO(eep): Was the previous implementation's behavior of merging all the
```

Comment:
better to match host behaviour (separate corpora) if possible. 

that said what are the semantics of put_data here? do the contents of each `corpus_dir` end up merged into `data/corpus`, do do they get copied as `data/corpus/corpus_dir_0`, `data/corpus/corpus_dir_1` etc... ?